pkg/authentication/handlers.go

```diff
@@ -136,12 +213,15 @@ func badRequest(w http.ResponseWriter, err error) {
 	return
 }
 
-func NewAPI(oauth2Context OAuth2ContextInterface, tracer trace.Tracer, logger logging.LoggerInterface) *API {
+func NewAPI(oauth2Context OAuth2ContextInterface, helper OAuth2HelperInterface, cookieManager AuthCookieManagerInterface, authCookiesTTL time.Duration, tracer trace.Tracer, logger logging.LoggerInterface) *API {
```

Comment:
```suggestion
func NewAPI(authCookiesTTL time.Duration, oauth2Context OAuth2ContextInterface, helper OAuth2HelperInterface, cookieManager AuthCookieManagerInterface, tracer trace.Tracer, logger logging.LoggerInterface) *API {
```

usually we put simpler objects first